Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+Nevertheless, currently the approach has several limitations:
+  - the same version for all packages from the repo should be used. Example: repo A depends on libraries B and C from runtime. The B and C versions should be identical or the source-build intermediate dependency will not be resolved.
+  - the source-build metadata for a specific repository can be declared only once. Example: repo A depends on librabries B and C from runtime. Only B or only C can have the source-build metadata for runtime, otherwise a duplicate dependency exception will be raised during build [GitHub issue](https://github.com/dotnet/source-build/issues/3003). 
+
+TODO - if product dependency is not part of source-build
+
+### Utilizing latest versions
+
+Use of latest versions for repository dependencies is the preffered and highly recommended approach in source-build setting.
+This stems from a mariad of reasons, the primary one being that in an end product build only the latest source-built version of any repository should exist.
+As a result, any older version of a repository would normally be overriden by the source-built one and down-stream repositories would have to utilize it.
+
+Deviation from this policy would result in higher maintanance cost for the source-build team, higher infrastructure load and bigger end product size.
+Nevertheless, if absolutely required, older versions of packages / libraries can be added to source-build throught a dedicated repository [source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages). 
+
+To have a package added to the repo, please [log an issue](https://github.com/dotnet/source-build/issues/new/choose) explaining the use-case in question and the source-build team will help you make the right decision.
+
+If the package you require is already in the source-build reference packages, it can be directly added to the `Version.Details.xml` dependency list:
+
+```xml
+  <Dependency Name="Microsoft.SourceBuild.Intermediate.source-build-reference-packages" Version="x.y.z">
+    <Uri>https://github.com/dotnet/source-build-reference-packages</Uri>
+    <Sha>xxx000</Sha>
+    <SourceBuildTarball RepoName="source-build-reference-packages" ManagedOnly="true" />
+  </Dependency>
+```
+
+## Dependencies without any code
+
+Dependencies that do not have any code (eg. SDKs with just props and targets) can usually be added to the [text-only packages](https://github.com/dotnet/source-build-reference-packages/tree/main/src/textOnlyPackages/src). 
+If the package in question is not already present in the list, please [log an issue](https://github.com/dotnet/source-build/issues/new/choose) to determine if a text-only package would solve the use-case in question.
```

Comment:
I think we should be able to document this so that devs can self-service.  They should be able to determine this on their own and utilize the SBRP tooling to generate any new text-only packages that are needed.

Reply:
Agreed, suggested to do this in [existing dependency documentation](https://github.com/dotnet/source-build/tree/main/Documentation/sourcebuild-in-repos?rgh-link-date=2022-11-22T19%3A50%3A11Z) as part of https://github.com/dotnet/source-build/pull/3083#discussion_r1029758078